test/e2e/controllerid_test.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2020 The prometheus-operator Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"context"
+	"testing"
+)
+
+func testMultipleOperatorsPrometheusServer(t *testing.T) {
+	testCtx := framework.NewTestCtx(t)
+	defer testCtx.Cleanup(t)
+	ns := framework.CreateNamespace(context.Background(), t, testCtx)
+	framework.SetupPrometheusRBAC(context.Background(), t, testCtx, ns)
+
+	name := "test-op-1"
+	p := framework.MakeBasicPrometheus(ns, name, name, 1)
+	_, err := framework.CreatePrometheusAndWaitUntilReady(context.Background(), ns, p)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	name = "test-op-2"
+	p = framework.MakeBasicPrometheus(ns, name, name, 1)
+	p.Annotations["operator.prometheus.io/controller-id"] = "42"
```

Comment:
(suggestion) let's create the 2 objects simultaneously, they will be reconciled in parallel and the test should run faster. Same comment for all other tests.

Reply:
how can I do it simultaneously? thanks

Reply:
I meant
1. create test-op-1
2. create test-op-2
3. wait for test-op-1 to be ready
4. wait for test-op-2 to be ready

Hopefully when we reach 4, the check shouldn't take long. But it requires to use `Create` +  `WaitForPrometheusReady` instead of  `CreatePrometheusAndWaitUntilReady`.